shell/platform/darwin/ios/framework/Headers/FlutterEngine.h

```diff
@@ -128,6 +133,10 @@ FLUTTER_EXPORT
  */
 - (BOOL)run;
 
+/** like run() method, but this method not block mainThread when init
+ * @param block callbackBlock, called when async init end. must not be nil!
+ */
+- (void)asyncRun:(InitCallBackBlock)block;
```

Comment:
Since you're adding an entirely new API, again for the sake of composable API and not mixing concepts together, you can actually just create a prepareVmAsync method or some such which gives you a callback when the Shell is created, at which point you can run `run` yourself. 

Then we don't have to multiplex all the variants of `run` with an async variant.